webapp/src/pages/offer/components/OfferComponent.js

```diff
@@ -0,0 +1,200 @@
+      const {offerList,userProfile} = this.state;
+      return(   
+      <div className="container"  align="center" >
+      <p className="h5 pt-5">We are pleased to offer you a place at the Deep Learning Indaba 2019. Please see the details of this offer below </p>
+
+      <form class="pt-5 ">
+        <p class="font-weight-bold">You have been accepted as a  {userProfile!=null ? userProfile.user_category: "<Category>"} </p>
+        <p class="font-weight-bold">Your status</p>
+        <div class="row">
+          <div class="col-8 font-weight-bold">
+            Travel Award:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.travel_award
+          : "Not available"}
+          </div>
+        </div>
+
+        <div class="row">
+          <div class="col-8 font-weight-bold">
+            Accommodation Award:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.accomodation_award
+          : "Not available"}
+          </div>
+        </div>
+
+        <div class="row pb-5 ">
+          <div class="col-8 font-weight-bold">
+            Payment Required:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.payment_required
+          : "Not available"}
+          </div>
+        </div>
+        <p class="font-weight-bold">Please accept or reject this offer by {offerList!=null ? offerList.expiry_date : "unable to load expiry date"} </p>
+        <div class="row">
+         <div class="col">
+          <button type="button" class="btn btn-danger" id="reject"onClick={
+            ()=>{
+              this.setState({
+                rejected:true,
+                showReasonBox:true
+              },
+              this.buttonSubmit());
+            }
+          } >
+              Reject
+          </button>
+          </div>
+          <div class="col">
+          <button type="button" class="btn btn-success" id="accept" onClick={()=>{
+              this.setState({
+                accepted:true
+              });
+              this.buttonSubmit();
+          }}>
+              Accept
+          </button>
+          </div>
+        </div> 
+
+        <div class="form-group">
+         { this.state.showReasonBox ? 
+          <div class="form-group mr-5  ml-5 pt-5" >
+          <textarea class="form-control pr-5 pl-10" id="textArea" onChange={()=>{this.handleChange(this.rejected_reason)}}  placeholder="Enter rejection message"></textarea>
+          <button type="button" class="btn" id="apply" onClick={
+            ()=>{
+              this.setState({
+                showReasonBox:false
+              },
+              this.buttonSubmit());
+            }
+            }>apply</button>
+        </div>
+        :""}
+        </div>
+      
+      </form>
+  </div>
+      )}
+
+    componentDidMount() {
+          this.getOffer();
+          profileService.getUserProfile(DEFAULT_EVENT_ID)
```

Comment:
getUserProfile takes a user_id, not event_id...

Reply:
I wasn't sure about loggedin user_id, so I pulled it from App.js componentDidMount()

Reply:
@LindanyRetroRabbit  What do you mean here?

Reply:
getUserProfile takes user_id parameter instead of event_id, so I should pass user_id as well.